### PR TITLE
Polaris table update api with tests

### PR DESCRIPTION
**Pull request by rsyncv**

### CODE REVIEW

File: metacat-connector-polaris/src/main/java/com/netflix/metacat/connector/polaris/PolarisConnectorTableService.java [hunk 1]
```diff
@@ -182,16 +188,57 @@ public TableInfo get(final ConnectorRequestContext requestContext, final Qualifi
...
+            final String newLoc = newTableMetadata.get(DirectSqlTable.PARAM_METADATA_LOCATION);
+            if (StringUtils.isBlank(prevLoc) || StringUtils.isBlank(newLoc)) {
+                final String message = String.format(
+                    "Invalid metadata for %s. Provided previous %s or new %s location is empty.",
+                    name, prevLoc, newLoc);
+                log.error(message);
+                throw new InvalidMetaException(name, message, null);
+            }
+            // optimistically attempt to update metadata location
+            final boolean updated = polarisConnector.updateTableMetadataLocation(
```
Comment (ajoymajumdar) [inline metacat-connector-polaris/src/main/java/com/netflix/metacat/connector/polaris/PolarisConnectorTableService.java:209]: Are we not storing other table properties?

Comment (ajoymajumdar) [inline metacat-connector-polaris/src/main/java/com/netflix/metacat/connector/polaris/PolarisConnectorTableService.java:209]: What happens if there is a timeout?

Comment (rsyncv) [reply metacat-connector-polaris/src/main/java/com/netflix/metacat/connector/polaris/PolarisConnectorTableService.java:209]: Not currently, we might start storing previous location as a separate field though.

Comment (rsyncv) [reply metacat-connector-polaris/src/main/java/com/netflix/metacat/connector/polaris/PolarisConnectorTableService.java:209]: We are planning to add a retry layer within the store layer, so this service layer does not need to handle those.

Comment (rsyncv) [reply metacat-connector-polaris/src/main/java/com/netflix/metacat/connector/polaris/PolarisConnectorTableService.java:209]: We'll have a sync to see if some of the other current use cases can be moved to use the Iceberg map instead.

**Review by raveeram — APPROVED** [review]

File: metacat-connector-polaris/src/main/java/com/netflix/metacat/connector/polaris/PolarisConnectorTableService.java [hunk 2]
```diff
@@ -182,16 +188,57 @@ public TableInfo get(final ConnectorRequestContext requestContext, final Qualifi
...
+                name.getDatabaseName(), name.getTableName(), prevLoc, newLoc);
+            // if succeeded then done, else try to figure out why and throw corresponding exception
+            if (updated) {
```
[the remark this answers is not in the document]
Comment (rsyncv) [reply metacat-connector-polaris/src/main/java/com/netflix/metacat/connector/polaris/PolarisConnectorTableService.java:212]: will do, making the change on the subsequent commit

**Pull request merged**